### PR TITLE
Issue #GEN-186: Added fixes to flow

### PR DESCRIPTION
**Pull request by tuj**
* Make progress bar clickable

### CODE REVIEW

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormAbout.php [hunk 1]
```diff
@@ -74,6 +79,8 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $this->store->set('body', $form_state->getValue('body'));
     $this->store->set('field_signup_required', $form_state->getValue('field_signup_required'));
 
+    $this->acceptStep(1);
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormAbout.php]: What is acceptStep 1 ?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php [hunk 2]
```diff
@@ -118,16 +119,68 @@ protected function getData() {
...
+          'title' => \Drupal::translation()->translate("Confirm"),
+          'href' => Url::fromRoute('itk_activity.multistep_confirm')->toString(),
+          'open' => $this->store->get('step5'),
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Accept a step in the form.
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php:173]: What do it mean to accept a step in the form?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormCategories.php [hunk 3]
```diff
@@ -29,6 +30,11 @@ public function getFormId() {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
 
+    // Set progress bar data.
+    $progressBar = $this->getProgressBar();
+    $progressBar['items'][2]['active'] = TRUE;
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormCategories.php]: Could the "2" here have been a name - categories?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormConfirm.php [hunk 4]
```diff
@@ -111,7 +112,16 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ],
     ];
 
+    // Set progress bar data.
+    $progressBar = $this->getProgressBar();
+    $progressBar['items'][5]['active'] = TRUE;
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormConfirm.php]: Could 5 have been a name?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormDetails.php [hunk 5]
```diff
@@ -29,6 +30,11 @@ public function getFormId() {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
 
+    // Set progress bar data.
+    $progressBar = $this->getProgressBar();
+    $progressBar['items'][4]['active'] = TRUE;
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormDetails.php]: Could 4 have been a name?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormImage.php [hunk 6]
```diff
@@ -30,6 +31,11 @@ public function getFormId() {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
 
+    // Set progress bar data.
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormImage.php]: This seam to be the same 3 lines of code only the number change... shoul it have been in an setFromProgress() function.

Could 3 have been a name?

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormInformation.php [hunk 7]
```diff
@@ -29,6 +30,11 @@ public function getFormId() {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form = parent::buildForm($form, $form_state);
 
+    // Set progress bar data.
+    $progressBar = $this->getProgressBar();
+    $progressBar['items'][1]['active'] = TRUE;
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormInformation.php]: Could 1 have been a name?

File: web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig [hunk 8]
```diff
@@ -1,10 +1,23 @@
 <div class="progress-bar">
```
Comment (cableman) [inline web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig:15]: Missing @file documentation.

And missing available vars: loop? item?

**Review by cableman — CHANGES_REQUESTED** [review]

File: web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig (hunk 8 above)
Comment (tuj) [reply web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig:15]: We do not have @file descriptions on the pattern lab twig files atm

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php (hunk 2 above)
Comment (tuj) [reply web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php:173]: Added extra comment

**Comment by tuj** [comment]
Ping for re-review

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php (hunk 2 above)
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php]: So what is this: https://github.com/aakb/genlydaarhus/blob/develop/web/themes/custom/genlyd_aarhus/templates/misc/status-messages.html.twig

File: web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig (hunk 8 above)
Comment (cableman) [reply web/themes/custom/genlyd_aarhus/source/_patterns/01-modules/progress-bar.twig:15]: So what is this: https://github.com/aakb/genlydaarhus/blob/develop/web/themes/custom/genlyd_aarhus/templates/misc/status-messages.html.twig

File: web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php [hunk 9]
```diff
@@ -175,12 +186,14 @@ protected function acceptStep($step) {
   protected function saveData() {
     $data = $this->getData();
 
+    $this->deleteStore();
+
+    // @TODO: Validate data
```
Comment (cableman) [inline web/modules/custom/itk_activity/src/Form/Multistep/MultistepFormBase.php]: Should it be validated?

**Review by cableman — APPROVED** [review]

**Pull request merged**